src/model/wallet.ts

```diff
@@ -257,11 +258,17 @@ export const walletInit = async (
   return { isNew, walletAddress };
 };
 
-export const loadWallet = async (
-  address?: EthereumAddress | undefined,
+export const loadWallet = async <S extends Screen>({
+  address,
   showErrorIfNotLoaded = true,
-  provider?: Provider
-): Promise<null | Wallet | LedgerSigner> => {
+  provider = web3Provider,
+  timeTracking,
+}: {
+  address: EthereumAddress | undefined;
```

Comment:
any reason not to keep the `?` optional on `address` so you don't have to pass the `address` in as undefined explicitly everywhere?

Reply:
Yeah was going to add that, let me do it